From ba8109f7402ff763cbed5607841d438aab56e2cf Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Sat, 5 Jan 2013 15:10:46 +0100
Subject: [PATCH 085/482] 	* grub-core/disk/diskfilter.c
 (grub_diskfilter_write): Call 	grub_error properly. 	*
 grub-core/disk/ieee1275/nand.c (grub_nand_write): Likewise. 	*
 grub-core/disk/loopback.c (grub_loopback_write): Likewise.

---
 ChangeLog                      | 7 +++++++
 grub-core/disk/diskfilter.c    | 3 ++-
 grub-core/disk/ieee1275/nand.c | 3 ++-
 grub-core/disk/loopback.c      | 3 ++-
 4 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index a28a1f7..f15e098 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2013-01-05  Vladimir Serbinenko  <phcoder@gmail.com>
+
+	* grub-core/disk/diskfilter.c (grub_diskfilter_write): Call
+	grub_error properly.
+	* grub-core/disk/ieee1275/nand.c (grub_nand_write): Likewise.
+	* grub-core/disk/loopback.c (grub_loopback_write): Likewise.
+
 2013-01-03  Vladimir Serbinenko  <phcoder@gmail.com>
 
 	* util/grub.d/10_kfreebsd.in: Correct the patch to zpool.cache as it's
diff --git a/grub-core/disk/diskfilter.c b/grub-core/disk/diskfilter.c
index ce4c706..4117b20 100644
--- a/grub-core/disk/diskfilter.c
+++ b/grub-core/disk/diskfilter.c
@@ -831,7 +831,8 @@ grub_diskfilter_write (grub_disk_t disk __attribute ((unused)),
 		 grub_size_t size __attribute ((unused)),
 		 const char *buf __attribute ((unused)))
 {
-  return GRUB_ERR_NOT_IMPLEMENTED_YET;
+  return grub_error (GRUB_ERR_NOT_IMPLEMENTED_YET,
+		     "diskfilter writes are not supported");
 }
 
 struct grub_diskfilter_vg *
diff --git a/grub-core/disk/ieee1275/nand.c b/grub-core/disk/ieee1275/nand.c
index ad30852..3474b3e 100644
--- a/grub-core/disk/ieee1275/nand.c
+++ b/grub-core/disk/ieee1275/nand.c
@@ -203,7 +203,8 @@ grub_nand_write (grub_disk_t disk __attribute ((unused)),
                  grub_size_t size __attribute ((unused)),
                  const char *buf __attribute ((unused)))
 {
-  return GRUB_ERR_NOT_IMPLEMENTED_YET;
+  return grub_error (GRUB_ERR_NOT_IMPLEMENTED_YET,
+		     "nand write is not supported");
 }
 
 static struct grub_disk_dev grub_nand_dev =
diff --git a/grub-core/disk/loopback.c b/grub-core/disk/loopback.c
index f3b19ef..fffd1bb 100644
--- a/grub-core/disk/loopback.c
+++ b/grub-core/disk/loopback.c
@@ -206,7 +206,8 @@ grub_loopback_write (grub_disk_t disk __attribute ((unused)),
 		     grub_size_t size __attribute ((unused)),
 		     const char *buf __attribute ((unused)))
 {
-  return GRUB_ERR_NOT_IMPLEMENTED_YET;
+  return grub_error (GRUB_ERR_NOT_IMPLEMENTED_YET,
+		     "loopback write is not supported");
 }
 
 static struct grub_disk_dev grub_loopback_dev =
-- 
1.8.2.1

